test/engine-luatest/gh_10847_memtx_sort_data_test.lua

```diff
@@ -0,0 +1,427 @@
+                    function(old, new, space, req, header, body)
+                        return new
+                    end)
+    ]]
+    cg.server:restart({
+        env = {
+            ['TARANTOOL_RUN_BEFORE_BOX_CFG'] = register_before_replace,
+        }
+    })
+
+    -- Now it's OK since the trigger is not called on recovery.
+    t.assert(cg.server:grep_log('Space \'test\' presorted indexes: done'))
+end
+
+-- Test that existence of vinyl spaces does not break the recovery.
+g_generic.test_vinyl_space_ignored = function(cg)
+    -- Create a vinyl space with data, save a snapshot and load it.
+    cg.server:exec(function()
+        local s = box.schema.create_space('test_vinyl', {engine = 'vinyl'})
+        s:create_index('pk')
+        s:create_index('sk')
+        for i = 1, 1000 do
+            s:insert({i})
+        end
+        box.snapshot()
+    end)
+    cg.server:restart()
+
+    -- We must have memtx space loaded using the sort data.
+    local memtx_log = 'Building presorted indexes in space \'test\'...'
+    local vinyl_log = 'Building presorted indexes in space \'test_vinyl\'...'
+    t.assert(cg.server:grep_log(memtx_log) ~= nil)
+    t.assert(cg.server:grep_log(vinyl_log) == nil)
+end
+
+-- Test recovery without sort data file if memtx_sort_data_enabled = true.
+g_generic.test_recover_without_sort_data = function(cg)
+    cg.server:exec(function()
+        -- Create a snapshot and sort data file.
+        box.snapshot()
+
+        -- Remove all sort data files.
+        local fio = require('fio')
+        local glob = fio.pathjoin(box.cfg.memtx_dir, '*.sortdata')
+        local files = fio.glob(glob)
+        t.assert(#files > 0) -- a sort data file must have been created
+        for _, file in pairs(files) do
+            fio.unlink(file)
+        end
+    end)
+
+    -- Recover without a sort data file using the regular SK build.
+    cg.server:restart()
+    t.assert(cg.server:grep_log('memtx_sort_data_enabled = true but' ..
+                                ' no memtx sort data file found'))
+    t.assert(cg.server:grep_log('Space \'test\': done'))
+    t.assert(cg.server:grep_log('entering the event loop'))
+end
+
+-- Test that the sort data file is recreated on box.snapshot after removal.
+g_generic.test_remove_sortdata_keep_snap = function(cg)
+    cg.server:exec(function()
+        -- Create a snapshot and sort data file.
+        box.snapshot()
+
+        -- Remove all sort data files.
+        local fio = require('fio')
+        local glob = fio.pathjoin(box.cfg.memtx_dir, '*.sortdata')
+        local files = fio.glob(glob)
+        t.assert(#files > 0) -- a sort data file must have been created
+        for _, file in pairs(files) do
+            fio.unlink(file)
+        end
+
+        -- Call box.snapshot() once again: sort data file recreated.
+        box.snapshot()
+        t.assert_equals(#fio.glob(glob), 1)
+    end)
+end
+
+-- Test that checkpoint fails if a sort data file exists already.
```

Comment:
Probably better to keep the existing `.sortdata` file as it's done for a `.snap` one.

Reply:
Done.